core/src/input_manager.cpp

```diff
@@ -0,0 +1,207 @@
+        m_holdKeyBindings[i] = false;
+    }
+
+    for (int i = 0; i < (int)MouseButtonCode::Last; ++i) {
+        m_mouseBindings[i] = false;
+        m_holdMouseBindings[i] = false;
+    }
+
+    for (int i = 0; i < (int)PadCode::Last; ++i) {
+        for (int j = 0; j < (int)PadButtonCode::Last; ++j) {
+            m_padBindings[i][j] = false;
+            m_holdKeyBindings[i] = false;
+        }
+
+        for (int j = 0; j < (int)JoyStickCode::Last; ++j) {
+            m_joyStickBindings[i][j] = 0.0f;
+        }
+    }
+}
+
+InputManager::~InputManager() {}
+
+/// Start Key Keyboard
+bool InputManager::key_pressed(KeyCode key) {
+    return m_keyBinding[(int)key];
+}
+
+bool InputManager::key_held_down(KeyCode key) {
+    return m_keyBinding[(int)key] && m_holdKeyBindings[(int)key];
+}
+
+bool InputManager::key_on_trigger(KeyCode key) {
+    return m_keyBinding[(int)key] && !m_holdKeyBindings[(int)key];
+}
+
+bool InputManager::key_on_release(KeyCode key) {
+    return m_keyBinding[(int)key] && !m_holdKeyBindings[(int)key];
+}
+
+/// End Of Keyboard
+
+/// Start Of Pad
+bool InputManager::pad_present(PadCode pad) {
+    return glfwJoystickPresent((int)pad);
+}
+
+bool InputManager::pad_button_pressed(PadCode pad, PadButtonCode button) {
+    return m_padBindings[(int)pad][(int)button];
+}
+
+bool InputManager::pad_button_held_down(PadCode pad, PadButtonCode button) {
+    return m_padBindings[(int)pad][(int)button] && m_holdPadBindings[(int)pad][(int)button];
+}
+
+bool InputManager::pad_button_triggered(PadCode pad, PadButtonCode button) {
+    return m_padBindings[(int)pad][(int)button] && !m_holdPadBindings[(int)pad][(int)button];
+}
+
+bool InputManager::pad_button_released(PadCode pad, PadButtonCode button) {
+    return m_padBindings[(int)pad][(int)button] && !m_holdPadBindings[(int)pad][(int)button];
+}
+
+float InputManager::get_pad_axis(PadCode pad, JoyStickCode joy_stick) {
+    return m_joyStickBindings[(int)pad][(int)joy_stick];
+}
+
+void InputManager::update_pads(GLFWwindow* glfwWindow) {
+    for (int i = 0; i < (int)PadCode::Last; i++) {
+        if (!pad_present((PadCode)i))
+            continue;
+
+        int count;
+        const float* axes = glfwGetJoystickAxes(i, &count);
+        if (!axes)
+            continue;
+
+        if (count > (int)JoyStickCode::Last)
+            count = (int)JoyStickCode::Last;
+        for (int n = 0; n < count; ++n)
+            m_joyStickBindings[i][n] = axes[n];
+
+        GLFWgamepadstate gameState;
+        if (glfwGetGamepadState(i, &gameState)) {
+            for (int j = 0; j < (int)PadButtonCode::Last; j++) {
+                bool padPressed = gameState.buttons[j];
+                m_padBindings[i][j] = padPressed;
+            }
+        } else if (glfwJoystickPresent(i)) {
+            const unsigned char* buttonPressed = glfwGetJoystickButtons(i, &count);
+            if (!buttonPressed)
+                continue;
+            for (int j = 0; j < count; ++j) {
```

Comment:
j never changes or is used as far as I can see so can remove the for loop part
Then I would change the bunch of ifs to a case statement
Someone else might have better suggestions @lmariscal

Reply:
ill talk to someone in uni about this